t/plugin/elasticsearch-logger.t

```diff
@@ -0,0 +1,447 @@
+#
+
+use t::APISIX 'no_plan';
+
+log_level('debug');
+repeat_each(1);
+no_long_string();
+no_root_location();
+no_shuffle();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if ((!defined $block->error_log) && (!defined $block->no_error_log)) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: sanity
+--- config
+    location /t {
+        content_by_lua_block {
+            local ok, err
+            local configs = {
+                -- full configuration
+                {
+                    endpoint_addr = "http://127.0.0.1:9200",
+                    field = {
+                        index = "services",
+                        type = "collector"
+                    },
+                    auth = {
+                        username = "elastic",
+                        password = "123456"
+                    },
+                    ssl_verify = false,
+                    timeout = 60,
+                    max_retry_count = 0,
+                    retry_delay = 1,
+                    buffer_duration = 60,
+                    inactive_timeout = 2,
+                    batch_max_size = 10,
+                },
+                -- minimize configuration
+                {
+                    endpoint_addr = "http://127.0.0.1:9200",
+                    field = {
+                        index = "services"
+                    }
+                },
+                -- property "endpoint_addr" is required
+                {
+                    field = {
+                        index = "services"
+                    }
+                },
+                -- property "field" is required
+                {
+                    endpoint_addr = "http://127.0.0.1:9200",
+                },
+                -- property "index" is required
+                {
+                    endpoint_addr = "http://127.0.0.1:9200",
+                    field = {}
+                }
+            }
+
+            local plugin = require("apisix.plugins.elasticsearch-logger")
+            for i = 1, #configs do
+                ok, err = plugin.check_schema(configs[i])
+                if err then
+                    ngx.say(err)
+                else
+                    ngx.say("passed")
+                end
+            end
+        }
+    }
+--- response_body_like
+passed
+passed
+property "endpoint_addr" is required
+property "field" is required
+property "field" validation failed: property "index" is required
+
+
+
+=== TEST 2: set route
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/elasticsearch-logger',
+                ngx.HTTP_DELETE,
+                 nil,
+                 [[{"action": "delete"}]]
+                 )
```

Comment:
```suggestion
            local code, body = t('/apisix/admin/plugin_metadata/elasticsearch-logger',
                ngx.HTTP_DELETE
           )
```

is ok, and we need to check if code >= 300

Reply:
```shell
➜  ~ curl http://127.0.0.1:9080/apisix/admin/plugin_metadata/elasticsearch-logger -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X DELETE
{"deleted":"1","key":"\/apisix\/plugin_metadata\/elasticsearch-logger","node":{},"action":"delete"}
➜  ~ curl http://127.0.0.1:9080/apisix/admin/plugin_metadata/elasticsearch-logger -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X DELETE
{"message":"Key not found"}
```
If the `apisix/admin/plugin_metadata/elasticsearch-logger` has be deleted before, we exec it again, it will return `404`, we does not care about the delete result, so we can not check`if code >= 300`

Reply:
we have `no_shuffle` on the boot of this test file, this means that the test cases will be executed in order. Why is there uncertainty that `apisix/admin/plugin_metadata/elasticsearch-logger` may or may not exist?

Reply:
> we have `no_shuffle` on the boot of this test file, this means that the test cases will be executed in order.
> 
> I think the existence of `apisix/admin/plugin_metadata/elasticsearch-logger` should be determined as far as the test file ` t/plugin/elasticsearch-logger.t` is concerned.
> 
> One CI test detail: before running a new test file each time, the test shell clears ETCD and initializes etcd with `apisix init_etcd` to prevent dirty data from other test files from affecting the current test file.

Yes, but it failed when I run `elasticsearch-logger.t` only:
```shell
prove -I../test_nginx/lib -r t/plugin/elasticsearch-logger.t
```
I think twice, maybe it is better to add the `DELETE` code, how about add a note to explain it?

Reply:
Perhaps you can add this to the documentation later: https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md

Reply:
> Perhaps you can add this to the documentation later: [https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md?rgh-link-date=2022-08-23T01%3A45%3A21Z)

Need we open a issue to discuss about it?